tests/jax/components/training/step_test.py

```diff
@@ -0,0 +1,475 @@
+                "network_agent_1": SimpleNamespace(
+                    params={"key": jnp.array([1.0, 1.0, 1.0])},
+                    network=SimpleNamespace(apply=apply),
+                ),
+                "network_agent_2": SimpleNamespace(
+                    params={"key": jnp.array([2.0, 2.0, 2.0])},
+                    network=SimpleNamespace(apply=apply),
+                ),
+            }
+        }
+        opt_states = {
+            "network_agent_0": 0,
+            "network_agent_1": 1,
+            "network_agent_2": 2,
+        }
+        store = SimpleNamespace(
+            dataset_iterator=iter([1, 2, 3]),
+            step_fn=step_fn,
+            timestamp=1657703548.5225394,  # time.time() format
+            trainer_parameter_client=MockParameterClient(),
+            trainer_counts={"next_sample": 2},
+            trainer_logger=MockTrainerLogger(),
+            sample_batch_size=2,
+            sequence_length=3,
+            trainer_agent_net_keys=trainer_agent_net_keys,
+            networks=networks,
+            gae_fn=gae_advantages,
+            opt_states=opt_states,
+            key=jax.random.PRNGKey(5),
+            num_minibatches=1,
+            epoch_update_fn=epoch_update,
+            num_epochs=2,
+        )
+        self.store = store
+
+
+class DummySample:
+    """Dummy Sample for step function from MAPGWITHTrustRegionStep component"""
+
+    def __init__(self) -> None:
+        self.data = SimpleNamespace(
+            observations={
+                "agent_0": OLT(
+                    observation=jnp.array(
+                        [
+                            [
+                                [0.1, 0.5, 0.7, 0.1, 0.5, 0.7],
+                                [0.1, 0.5, 0.7, 0.1, 0.56, 0.87],
+                            ],
+                            [
+                                [0.1, 0.5, 0.7, 0.1, 0.5, 0.7],
+                                [0.1, 0.5, 0.7, 0.13, 0.5, 0.78],
+                            ],
+                        ]
+                    ),
+                    legal_actions=jnp.array(
+                        [
+                            [
+                                [1, 1, 1, 1, 0, 0],
+                                [0, 1, 1, 0, 1, 0],
+                            ],
+                            [
+                                [0, 0, 0, 1, 0, 1],
+                                [1, 1, 1, 0, 1, 0],
+                            ],
+                        ]
+                    ),
+                    terminal=jnp.array(
+                        [
+                            [
+                                [0, 0, 0, 0, 0, 1],
+                                [0, 1, 0, 0, 0, 0],
+                            ],
+                            [
+                                [0, 1, 1, 0, 1, 1],
+                                [0, 0, 0, 1, 1, 0],
+                            ],
+                        ]
+                    ),
+                ),
```

Comment:
From what I have seen from the debug environment samples it seems that `terminal` should instead take on this form 
```
terminal=jnp.array([
                           [[0.],
                            [0.],
                            [0.]],
                            [[0.],
                             [0.],
                             [0.]]
                      ], )),
```